source/codegen/common_helpers.py

```diff
@@ -304,6 +304,8 @@ def get_size_mechanism(parameter):
 
 def get_size_expression(parameter):
     size_mechanism = get_size_mechanism(parameter)
+    if 'size' not in parameter:
```

Comment:
note: I added this to give a better error message when an out array parameter is missing its size.